src/ConnectedMode/Binding/ServerConnectionConfigurationProvider.cs

```diff
@@ -0,0 +1,109 @@
+
+namespace SonarLint.VisualStudio.ConnectedMode.Binding
+{
+    [Export(typeof(IServerConnectionConfigurationProvider))]
+    [PartCreationPolicy(CreationPolicy.Shared)]
+    internal class ServerConnectionConfigurationProvider : IServerConnectionConfigurationProvider
+    {
+        private readonly ISolutionBindingRepository solutionBindingRepository;
+        private readonly IThreadHandling threadHandling;
+        private readonly IConnectionIdHelper connectionIdHelper;
+
+        [ImportingConstructor]
+        [ExcludeFromCodeCoverage]
+        public ServerConnectionConfigurationProvider(ISolutionBindingRepository solutionBindingRepository)
+            : this(solutionBindingRepository, ThreadHandling.Instance, new ConnectionIdHelper())
+        {
+        }
+
+        internal ServerConnectionConfigurationProvider(ISolutionBindingRepository solutionBindingRepository, IThreadHandling threadHandling, IConnectionIdHelper connectionIdHelper)
+        {
+            this.solutionBindingRepository = solutionBindingRepository;
+            this.threadHandling = threadHandling;
+            this.connectionIdHelper = connectionIdHelper;
+        }
+
+        public IEnumerable<T> GetServerConnectionConfiguration<T>() where T : ServerConnectionConfiguration
+        {
+            threadHandling.ThrowIfOnUIThread();
+
+            var bindingList = new List<ServerConnectionConfiguration>();
+
+            var bindings = solutionBindingRepository.List().Distinct(new BoundSonarQubeProjectUriComparer());
+
+            foreach (var binding in bindings)
+            {
+                var connectionID = connectionIdHelper.GetConnectionIdFromUri(binding.ServerUri, binding.Organization?.Key);
+
+                if (binding.ServerUri == ConnectionIdHelper.SonarCloudUri)
+                {
+                    bindingList.Add(new SonarCloudConnectionConfigurationDto(connectionID, true, binding.Organization?.Key));
+                }
+                else
+                {
+                    bindingList.Add(new SonarQubeConnectionConfigurationDto(connectionID, true, binding.ServerUri.ToString()));
+                }
+            }
+
+            return bindingList.OfType<T>();
+        }
+
+        private sealed class BoundSonarQubeProjectUriComparer : IEqualityComparer<BoundSonarQubeProject>
+        {
+            public bool Equals(BoundSonarQubeProject x, BoundSonarQubeProject y)
+            {
+                if (ReferenceEquals(x, y))
+                {
+                    return true;
+                }
+
+                if (x is null && y is null)
+                {
+                    return true;
+                }
+
+                if (x is null || y is null)
+                {
+                    return false;
+                }
+
+                return x.ServerUri.Equals(y.ServerUri);
+            }
+
+            public int GetHashCode(BoundSonarQubeProject obj)
+            {
+                return obj.ServerUri.GetHashCode();
```

Comment:
You check for null in Equals, but not here

Reply:
We don't need to here it's the expected behaviour.